app/src/org/commcare/views/widgets/WidgetUtils.java

```diff
@@ -75,4 +78,22 @@ public static Intent createScanIntent(Context context, String... formats) {
         }
         return intentIntegrator.createScanIntent();
     }
+
+    @SuppressLint("InlinedApi")
+    public static Intent createPickMediaIntent(Context context, String mimeType) {
+        Intent intent = new Intent();
+        if (isPhotoPickerSupported(context)) {
+            intent.setAction(MediaStore.ACTION_PICK_IMAGES);
+        } else {
+            intent.setAction(Intent.ACTION_GET_CONTENT);
+        }
+        return intent.setType(mimeType);
+    }
+
+    @SuppressLint("InlinedApi")
+    public static boolean isPhotoPickerSupported(Context context) {
+        Intent intent = new Intent(MediaStore.ACTION_PICK_IMAGES);
```

Comment:
How is this passing on versions older than api 33 as this field is [not present there](https://developer.android.com/reference/android/provider/MediaStore#ACTION_PICK_IMAGES) ?

Reply:
This puzzled me a bit, but there is a Google project called [Mainline](https://source.android.com/docs/core/ota/modular-system) that allows new functionality and libraries to be delivered outside of the normal Android release cycle, making it possible to provide newer functionality to older versions of Android, here's the list of [modules](https://source.android.com/docs/core/ota/modular-system#available-modules). As per the `ACTION_PICK_IMAGES` itself, considering that it is a constant, during build the reference is replaced by its value, so it should be safe.